corehq/apps/fixtures/download.py

```diff
@@ -150,7 +150,8 @@ def empty_padding_list(length):
         max_groups = 0
         max_locations = 0
         max_field_prop_combos = {field_name: 0 for field_name in data_type.fields_without_attributes}
-        fixture_data = FixtureDataItem.by_data_type(domain, data_type.get_id)
+        fixture_data = sorted(FixtureDataItem.by_data_type(domain, data_type.get_id),
```

Comment:
This wouldn't ensure idempotency if the user doesn't upload the fixtures in the order, would it?

Also, this would probably only fix the order in the download and not the way the options appear on the multiple choice question, no? Or will that be fixed with this?

Reply:
> This wouldn't ensure idempotency if the user doesn't upload the fixtures in the alphabetical order, would it?

It ensures that the fixture is downloaded in the predefined sorted order, alphabetization should have nothing to do with it. That is why we sort on `sort_key` not something like name. That sort key is set by the upload. So the upload and download would be idempotent.

> Also, this would probably only fix the order in the download and not the way the options appear on the multiple choice question, no? Or will that be fixed with this?

There is no bug with the order things are appearing in multiple choice questions as far as I know. That is showing correct behavior and this does not change that. This only changes download order to match the order items appear in questions, because that order is set by upload, and we want a download followed by an upload to remain idempotent.